core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/diagnostics/handler/AbstractDiagnosticsOperationHandler.java

```diff
@@ -0,0 +1,34 @@
+package com.sequenceiq.cloudbreak.core.flow2.diagnostics.handler;
+
+import com.cloudera.thunderhead.service.common.usage.UsageProto;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.core.flow2.diagnostics.DiagnosticsFlowException;
+import com.sequenceiq.cloudbreak.core.flow2.diagnostics.event.DiagnosticsCollectionEvent;
+import com.sequenceiq.cloudbreak.core.flow2.diagnostics.event.DiagnosticsCollectionFailureEvent;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+public abstract class AbstractDiagnosticsOperationHandler extends ExceptionCatcherEventHandler<DiagnosticsCollectionEvent> {
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<DiagnosticsCollectionEvent> event) {
+        try {
+            return executeOperation(event);
+        } catch (Exception e) {
+            throw new DiagnosticsFlowException(String.format("Error during diagnostics operation: %s", getOperationName()), e);
+        }
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<DiagnosticsCollectionEvent> event) {
+        return new DiagnosticsCollectionFailureEvent(resourceId, e, event.getData().getResourceCrn(), event.getData().getParameters(), getFailureType().name());
+    }
```

Comment:
i would like to catch every error during the diagnostics steps (as it's happening now), the failure type is related for the step type not for the exception type (as it would probably not the default usage of ExceptionCatcherHandler), so make sense to not let throw any exception in ExceptionCatcherHandler (but use it this way here in these abstract classes)

Reply:
i will close that for now, will revisit the changes probably in a follow up commit if it will be needed